chainer/functions/pooling/spatial_pyramid_pooling_2d.py

```diff
@@ -73,9 +79,14 @@ def spatial_pyramid_pooling_2d(x, pyramid_height, pooling_class):
         ksize = (ksize_h, ksize_w)
         pad = (pad_h, pad_w)
 
-        if pooling_class is chainer.functions.MaxPooling2D:
-            pooler = pooling_class(ksize=ksize, stride=None, pad=pad,
-                                   cover_all=True)
+        if pooling_class is not None:
```

Comment:
How about checking only one of `pooling_class` or `pooling` is specified?

Reply:
Fixed. I added a `ValueError` some accompanying tests.